src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1649,6 +1706,67 @@ public Pattern getPattern()
         }
     }
 
+    /**
+     * Enum representing the flags on a Message.
+     * <p>
+     * Note: The Values defined in this Enum are not considered final and only represent the current State of <i>known</i> Flags.
+     */
+    enum MessageFlag
+    {
+        /**
+         * The Message has been published to subscribed Channels (via Channel Following)
+         */
+        CROSSPOSTED(0),
+        /**
+         * The Message originated from a Message in another Channel (via Channel Following)
+         */
+        IS_CROSSPOST(1),
+        /**
+         * Embeds are suppressed on the Message.
+         * @see net.dv8tion.jda.api.entities.Message#isSuppressedEmbeds() Message#isSuppressedEmbeds()
+         */
+        EMBEDS_SUPPRESSED(2),
+        /**
+         * Indicates, that the source message of this crosspost was deleted.
+         * This should only be possible in combination with {@link #IS_CROSSPOST}
+         */
+        SOURCE_MESSAGE_DELETED(3),
+        /**
+         * Indicates, that this Message came from the urgent message system
+         */
+        URGENT(4);
+
+        private final int offset;
+
+        MessageFlag(int offset)
+        {
+            this.offset = offset;
```

Comment:
This should store the raw value, similar to permissions. There is no need to caclulate this every time we need it.